### PR TITLE
[4.0] TagsHelper parameterized queries

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue # .

### Summary of Changes

Adds parameterized queries in `Joomla\CMS\Helper\TagsHelper`.

### Testing Instructions

Test that adding and remove tags from content items work.
Test that creating tags on the fly and searching tags work.
Test that `com_tags` frontend views like before.

### Expected result

Works like before.

### Documentation Changes Required

No.

### CODE REVIEW

File: libraries/src/Helper/TagsHelper.php [hunk 1]
```diff
@@ -826,47 +871,64 @@ public static function searchTags($filters = array())
...
+				->bind(':title', $filters['title']);
 		}
 
 		// Filter on the published state
 		if (isset($filters['published']) && is_numeric($filters['published']))
 		{
-			$query->where('a.published = ' . (int) $filters['published']);
+			$published = (int) $filters['published'];
+			$query->where($db->quoteName('a.published') . ' = :published')
+				->bind(':published', $filters['published'], ParameterType::INTEGER);
```
Comment (Quy) [inline libraries/src/Helper/TagsHelper.php]:
```suggestion
				->bind(':published', $published, ParameterType::INTEGER);
```

**Comment by Quy** [comment]
Edit an article.
Type an existing tag in the Tags field.
No results match.

> Undefined variable: filter in \libraries\src\Helper\TagsHelper.php on line 900

**Comment by SharkyKZ** [comment]
@Quy Fixed.

**Comment by Quy** [comment]
When creating a new tag while editing an article:

> Save failed with the following error: Incorrect datetime value: 'CURRENT_TIMESTAMP()' for column `joomla_db`.`rkl56_contentitem_tag_map`.`tag_date` at row 1

**Comment by SharkyKZ** [comment]
Should be fixed now.

**Comment by Quy** [comment]
> PHP Warning:  array_merge(): Expected parameter 3 to be an array, string given in \libraries\src\Helper\TagsHelper.php on line 109

> PHP Warning:  implode(): Invalid arguments passed in \libraries\src\Helper\TagsHelper.php on line 109

File: libraries/src/Helper/TagsHelper.php [hunk 2]
```diff
@@ -335,23 +343,28 @@ public function deleteTagData(TableInterface $table, $contentItemId)
...
+				[
+					$db->quoteName('m.type_alias') . ' = :contentType',
+					$db->quoteName('m.content_item_id') . ' = :id',
 					$db->quoteName('t.published') . ' = 1',
-				)
-			);
+				]
+			)
+			->bind(':contentType', $contentType)
+			->bind(':id', $id);
```
Comment (Quy) [inline libraries/src/Helper/TagsHelper.php]:
```suggestion
			->bind(':id', $id, ParameterType::INTEGER);
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 8b1d50bd1fe074ff7219e173f8cbe94acbe2db25<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27107">issues.joomla.org/tracker/joomla-cms/27107</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 8b1d50bd1fe074ff7219e173f8cbe94acbe2db25<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27107">issues.joomla.org/tracker/joomla-cms/27107</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27107">issues.joomla.org/tracker/joomla-cms/27107</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**